plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -251,13 +317,87 @@ def remove_units(self, repo, units, config):
         :param config: plugin configuration
         :type  config: pulp.plugins.config.PluginCallConfiguration
         """
-        repo_manager = manager_factory.repo_manager()
-        scratchpad = repo_manager.get_repo_scratchpad(repo.id)
-        tags = scratchpad.get(u'tags', [])
-        unit_ids = set([unit.unit_key[u'image_id'] for unit in units])
-        for tag_dict in tags[:]:
-            if tag_dict[constants.IMAGE_ID_KEY] in unit_ids:
-                tags.remove(tag_dict)
+        self._purge_unreferenced_tags(repo, units)
+        self._purge_orphaned_blobs(repo, units)
+
+    @staticmethod
+    def _purge_unreferenced_tags(repo, units):
+        """
+        Purge tags associated with images in the repository.
 
+        :param repo: The affected repository.
+        :type  repo: pulp.plugins.model.Repository
+        :param units: List of removed units.
+        :type  units: list of: pulp.plugins.model.AssociatedUnit
+        """
+        unit_ids = set()
+        manager = manager_factory.repo_manager()
+        scratchpad = manager.get_repo_scratchpad(repo.id)
+        for unit in units:
+            if unit.type_id != Image.TYPE_ID:
+                continue
+            unit_ids.add(unit.unit_key[u'image_id'])
+        tags = scratchpad.get(u'tags', [])
+        for tag in tags[:]:
```

Comment:
That is what the code did before I rewrote it.  Not sure why the list was being copied so left it.  I don't see any value in copying it.